core/src/main/java/org/wso2/msf4j/internal/router/HttpMethodInfo.java

```diff
@@ -323,4 +323,12 @@ public void end() throws Exception {
     public boolean isStreamingSupported() {
         return httpStreamHandler != null;
     }
+
+    public Object[] getArgs() {
+        return Arrays.copyOf(args, args.length);
```

Comment:
Is it absolutely necessary to create a copy of the array here? If not better not to do it as it may affect memory consumption.